website/static/js/filepage/index.js

```diff
@@ -95,9 +95,40 @@ var FileViewPage = {
         self.file = self.context.file;
         self.node = self.context.node;
         self.editorMeta = self.context.editor;
-        //Force canEdit into a bool
-        self.canEdit = m.prop(!!self.context.currentUser.canEdit);
-
+        self.file.checkoutUser = null;
+        self.request_done = false;
+        self.isCheckoutUser = function() {
+            $.ajax({
+                method: 'get',
+                url: window.contextVars.apiV2Prefix + 'files' + self.file.path + '/',
+                beforeSend: $osf.setXHRAuthorization
+            }).done(function(resp) {
+                self.request_done = true;
+                self.file.checkoutUser = resp.data.relationships.checkout.links.related.href ? ((resp.data.relationships.checkout.links.related.href).split('users/')[1]).replace('/', ''): null;
+                if ((self.file.checkoutUser) && (self.file.checkoutUser !== self.context.currentUser.id)) {
+                    m.render(document.getElementById('alertBar'), m('.alert.alert-warning[role="alert"]', m('span', [
+                        m('strong', 'File is checked-out.'),
+                        'This file has been checked-out by a ',
+                        m('a[href="/' + self.file.checkoutUser + '"]', 'collaborator'),
+                        '. It beeds to be checked back in before any changes can be made.'
+                    ])));
+                }
+            }).fail(function(resp) {
+                $osf.growl('Error', 'Unable to get check-out status of this file.');
```

Comment:
Again, is this an expected or unexpected error? How should the error be presented in either case? Does it need to be logged to Sentry?

I'm not sure showing this growl error is appropriate. Is it important that the user be notified about this failure?

I noticed this error because my waterbutler server was not up-to-date, but this could happen in a production situation  if there's a bug somewhere. I got the "Unable to get checkout-out status of the file." upon visiting the file detail page. As a user, I would not know what this error means (I would never have heard of the "checkout" feature).